PCF-ADR-0005-mcad-api-groups-migration.md

```diff
@@ -0,0 +1,131 @@
+
+### Phase 2: Progressive Migration
+
+The migration should be announced on the usual communication channels.
+
+The following downstream projects must be migrated to using the new API groups:
+
+* The TorchX MCAD scheduler: https://github.com/pytorch/torchx/blob/main/torchx/schedulers/kubernetes_mcad_scheduler.py
+* The CodeFlare SDK: https://github.com/project-codeflare/codeflare-sdk
+* ODH Data Science Pipeline operator: https://github.com/opendatahub-io/data-science-pipelines-operator
+* KubeRay documentation: https://ray-project.github.io/kuberay/guidance/kuberay-with-MCAD (no mention of the AppWrapper API group, but the links to the project materials can be updated)
+
+### Phase 3: Decommission old API Groups
+
+Once all the downstream projects have migrated, and existing users acknowledged the migration plan, the following clean-up tasks should be performed:
+
+* Delete the old MCAD API packages
+* Delete the old MCAD client packages
+* Delete the old MCAD CRDs
+* Remove RBAC for the old API groups from MCAD manifests (both in Helm charts and in the CodeFlare operator)
+* Delete the duplicated MCAD controllers
+* Delete the duplicated InstaScale controllers
+* Remove RBAC for the old API groups from InstaScale manifests
+
+## Open Questions
+
+* Is dual mode support necessary for the quota APIs?
+  While they'll eventually be public APIs, these are rather new, and may not be actually used yet.
+
+## Alternatives
+
+Given the TorchX MCAD scheduler is currently delivered as part of the [CodeFlare fork of TorchX](https://github.com/project-codeflare/torchx), all the impacted components are managed internally.
+That means a _one-shot_ migration can be a viable alternative, assuming we accept the development branches of these components may transiently break, within the  span of that _one-shot_ migration release cycle.
+
+As an example, this _one-shot_ migration could be achieved during the next development cycle of the CodeFlare stack, i.e., the upcoming v0.7.0 release, in the following order:
+
+| Component        | Version |
+| ---------------- | ------- |
+| MCAD             | v1.34.0 |
+| InstaScale       | v0.7.0  |
+| CodeFlare TorchX | v0.7.0  |
+| CodeFlare SDK    | v0.7.0  |
+
+The ODH Data Science Pipeline operator update can be done as soon as ODH upgrades to that newer CodeFlare version.
+
+The KubeRay documentation can be updated independently.
+
+## Stakeholder Impacts
+
+| Group                         | Key Contacts       | Date       | Impacted? |
+| ----------------------------- | ------------------ | ---------- | --------- |
+| CodeFlare SDK                 | Mustafa Eyceoz     |            | yes       |
+| MCAD                          | Abhishek Malvankar |            | yes       |
+| InstaScale                    | Abhishek Malvankar |            | yes       |
+| CodeFlare Operator            | Anish Asthana      |            | no        |
```

Comment:
Can you add me for the KubeRay apiserver and KubeRay documentation ? If the implementation of this change will trail, there will be an impact on the kuberayapi server. 

TL;DR. Kueberay api server will submit appwrappers for the KubeRay CRDS. We plan on depending on the MCAD api package to programately create the appwrappers.

Reply:
Added with 5473484368460890188cba3e732ef9ffcaa9606f.